### ISSUE TITLE
ghc.exe: addLibrarySearchPath error

### ISSUE THREAD

**Issue by vagarenko** [haskell/cabal#4741]
Originally posted here: https://ghc.haskell.org/trac/ghc/ticket/14183

My OS is Windows 7. Recently I've removed Haskell Platform 8.0.1 and installed new Platform 8.2.1

Since then, when compiling a module GHC sometimes spits the following messages:

```
ghc.exe: addLibrarySearchPath: C:\Program Files\Haskell Platform\8.0.1\mingw\lib (Win32 error 3): The system cannot find the path specified.
```

dozens of times in a row. But the compilation succeeds.
The directory `C:\Program Files\Haskell Platform\8.0.1\` doesn't exist on my machine. 
I don't have any references to `C:\Program Files\Haskell Platform\8.0.1\` in my cabal.config file.

I haven't figured out when exactly this happens but it seems to happen only when the module has Template Haskell splices in it.

**UPDATE:**
If I run `cabal haddock` the error changes to:

```
haddock.exe: addLibrarySearchPath: C:\Program Files\Haskell Platform\8.0.1\mingw\lib (Win32 error 3): The system cannot find the path specified.
```
but the documentation is created.

**Comment by 23Skidoo** [comment]
Check your `%APPDATA%/cabal/config` file.

**Comment by vagarenko** [comment]
Turns out it isn't a Cabal bug.
https://ghc.haskell.org/trac/ghc/ticket/14183

**Comment by 23Skidoo** [comment]
tl;dr: solved by recreating the sandbox.
